src/Microsoft.OData.ModelBuilder/Conventions/Attributes/NullableContextAttributeEdmTypeConvention.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.OData.ModelBuilder.Conventions.Attributes
+{
+    /// <summary>
+    /// Configures properties of a Type that has the NullableContextAttribute.
+    /// </summary>
+    /// <remarks>This MUST come after NullableAttribute</remarks>
+    internal class NullableContextAttributeEdmTypeConvention : AttributeEdmTypeConvention<StructuralTypeConfiguration>
+    {
+        //
+        // For the interpretation of nullability metadata, see
+        // https://github.com/dotnet/roslyn/blob/master/docs/features/nullable-metadata.md
+        //
+
+        private const string NullableContextAttributeFullName = "System.Runtime.CompilerServices.NullableContextAttribute";
+
+        public NullableContextAttributeEdmTypeConvention()
+            : base(attribute => attribute.GetType().FullName == NullableContextAttributeFullName, allowMultiple: false)
+        {
+        }
+
+        /// <summary>
+        /// Configures the NonNullablity of a type's properties based on the NullableContext.
+        /// </summary>
+        /// <param name="edmTypeConfiguration">The edm type being configured.</param>
+        /// <param name="attribute">The <see cref="Attribute"/> found on the property.</param>
+        /// <param name="model">The ODataConventionModelBuilder used to build the model.</param>
+        public override void Apply(StructuralTypeConfiguration edmTypeConfiguration,
+                                   ODataConventionModelBuilder model,
+                                   Attribute attribute)
+        {
+            Contract.Assert(edmTypeConfiguration != null);
+
+            //
+            // Only propigate Nullability if flag == 1 (not annotated)
+            //
```

Comment:
shall we only apply this convention to the non-added explicitly structural type?

Reply:
See previous comments. The Nullable attributes are only added by the compiler (internal only) and don't affect runtime behavior. Developers cannot use the attribute as its class is not defined in any runtime. So explicit isn't a factor.